src/GraphQL/Server.hs

```diff
@@ -181,33 +181,54 @@ instance forall v. ReadValue v => ReadValue (Maybe v) where
 -- name doesn't match.
 data NamedFieldExecutor m = NamedFieldExecutor AST.Name (m GValue.Value)
 
+executeNamedField :: Monad m => AST.Field -> NamedFieldExecutor m -> m (Maybe GValue.ObjectField)
+executeNamedField (AST.Field alias name _ _ _) (NamedFieldExecutor k mValue)
+  | name == k = do
+      value <- mValue
+      let name' = GValue.unsafeMakeName $ if alias == "" then name else alias
+      pure (Just (GValue.ObjectField name' value))
+  | otherwise = pure Nothing
+
+-- Deconstruct object type signature and handler value at the same
+-- time and run type-directed code for each field.
+resolveField :: forall a (m :: * -> *). BuildFieldResolver m a => m GValue.ObjectField -> FieldHandler m a -> AST.Selection -> m GValue.ObjectField
```

Comment:
Ahhhh, so _that's_ what `Type` means. I'm familiar with `*` from the repl & compilation errors, and had assumed that `Type` was something special related to all the magic you're doing. :smiling_imp: 

Changed to use `Type`.

FWIW, this signature was added with a keystroke, using the 'apply suggestion' feature of intero, which looks at the compiler warning/error and applies the suggestion.